### PR TITLE
Update Faraday Gem Version [#178090958]

### PR DESCRIPTION
**Pull request by karthik-sivadas**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'faraday', '>= 0.8', '~> 0.14.0'
+gem 'faraday', '>= 1.0'
```
Comment (itsgg) [inline Gemfile]: Use pessimistic locking ~> 1.0

File: clever_tap.gemspec [hunk 2]
```diff
@@ -25,7 +25,7 @@ Gem::Specification.new do |spec|
       'public gem pushes.'
   end
 
-  spec.add_dependency 'faraday', '>= 0.8', '<= 0.14.0'
+  spec.add_dependency 'faraday', '>= 1.0'
```
Comment (itsgg) [inline clever_tap.gemspec]: Use pessimistic locking ~> 1.0

File: lib/clever_tap/client.rb [hunk 3]
```diff
@@ -33,7 +33,7 @@ def connection
         configure.call(config)
 
         # NOTE: set adapter only if there isn't one set
-        config.adapter :net_http if config.builder.handlers.empty?
+        config.adapter :net_http if config.builder.adapter.nil?
```
Comment (itsgg) [inline lib/clever_tap/client.rb:36]: Why is this change needed?

Comment (karthik-sivadas) [reply lib/clever_tap/client.rb:36]: > Why is this change needed?

After updating the version we get the adapter details in the adapter inside RackBuilder instead of handlers.

**Review by itsgg — APPROVED** [review]
LGTM
